qadence/types.py

```diff
@@ -370,3 +373,12 @@ class OpName(StrEnum):
     """The wait operation."""
     PROJ = "Projector"
     """The projector operation."""
+
+
+class Engine(StrEnum):
+    TORCH = "torch"
+    JAX = "jax"
+
+
+ParamDictType = dict[str, ArrayLike]
+DifferentiableExpression = Callable[..., ArrayLike]
```

Comment:
i like having type aliases like this since it makes the code more readable. i could also just use "Callable" if you prefer.

i introduced it to be able to type:
```
    embeddings: dict[sympy.Expr, DifferentiableExpression] = {
        expr: make_differentiable(expr=expr, engine=engine)
        for expr in unique_expressions
        if not expr.is_number
    }
```
but again, i can just do "Callable" (which is not really correct right)

Reply:
i am happy to hear about alternative aliases. `DifferentiableFunction` is weird